File: GameEngine_IOS/GameEngine/Models/Camera.m
#import "Camera.h"

@implementation Camera

/**
 Initiator of camera
 *
 */
- (id)init {
    self = [super init];
    if (self) {
        self -> _position = [[Vector3f alloc] init: 0.0f : 1.0f : 0.0f];
        self -> _yaw = -120;
    }
    return self;
}



/**
 * Read the keys pressed by the used and updates the
 * position of the camera
 */
- (void) move {
    double pitchR = GLKMathDegreesToRadians([self pitch]);
    double yawR = GLKMathDegreesToRadians([self yaw]);
    
    //if(Keyboard.isKeyDown(Keyboard.KEY_UP)) {
    _position.x += (-STEP * sin(yawR));
    _position.y += (+STEP * sin(pitchR));
    _position.z += (-STEP * sin(pitchR)) + (-STEP * cos(yawR));
    //}
    //if(Keyboard.isKeyDown(Keyboard.KEY_DOWN)) {
    /*_position.x += (+STEP * sin(yawR));
    _position.y += (-STEP * sin(pitchR));
    _position.z += (+STEP * sin(pitchR)) + (+STEP * cos(yawR));*/
    //}
}


/**
 * Rotate the camera that the user sees
 */
- (void) rotate {
    /*Looks down*/
    //if(Keyboard.isKeyDown(Keyboard.KEY_Z)) {
    //_pitch -= ANGLE_OF_ROTATION;
    //}
    
    /*Looks up*/
    //if(Keyboard.isKeyDown(Keyboard.KEY_W)) {
    //_pitch += ANGLE_OF_ROTATION;
    //}
    
    /*Looks left*/
    //if(Keyboard.isKeyDown(Keyboard.KEY_A)) {
    //_yaw += ANGLE_OF_ROTATION;
    //}
    
    /*Looks right*/
    //if(Keyboard.isKeyDown(Keyboard.KEY_S)) {
    //_yaw -= ANGLE_OF_ROTATION;
}

/**
 Unassign the associated values of the class
 */
- (void) dealloc {
    self -> _position = nil;
}

@end

File: GameEngine_IOS/GameEngine/Models/Camera.swift
import Foundation
import GLKit



/**
* Represents the the camera used by the user to see the
* 3D World
*/
open class Camera
{

    
    
    /**
    * The position where the camera is
    */
    var position : Vector3f;
    
    /**
    * Pitch is up and down (rotation around X-axis)
    */
    var pitch : Float;
    
    /**
    * Yaw is the angle when moving the head left and right (rotation around Y-axis)
    */
    var yaw : Float;
    
    /**
    * Roll, which we usually don't experience is when you tilt your head (rotation around Z-axis)
    */
    var   roll : Float;
    
    /**
    Initiator of camera
    *
    */
    public init() {
        self.position = Vector3f(x: 0.0, y: 1.0, z: 0.0);
        self.pitch = 0;
        self.yaw = -90;
        self.roll = 0
    }
}

